File: src/Tests/QueryCategorizerTest.java
package Tests;

import Phase02.QueryCategorizer;

import Phase02.QueryKeeper;
import org.junit.*;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.Mockito;
import org.mockito.runners.MockitoJUnitRunner;

import java.io.ByteArrayInputStream;
import java.util.HashSet;
import java.util.Scanner;

@RunWith(MockitoJUnitRunner.class)
public class QueryCategorizerTest {

    @Mock
    private Scanner mockedScanner;

    private QueryCategorizer queryCategorizer;

    @Before
    public void setup(){
        queryCategorizer = new QueryCategorizer();
        String query = "mom dad +sister +sisi -brother -bro";
        System.setIn(new ByteArrayInputStream(query.getBytes()));
    }

    @Test
    public void testCategorizeForNoSignQueries() {
        QueryKeeper queryKeeper = queryCategorizer.categorizeQuery();
        Assert.assertArrayEquals(new String[]{"dad","mom"},
                queryKeeper.getWithOutSign().toArray());

    }

    @Test
    public void testCategorizeForPlusQueries() {
        QueryKeeper queryKeeper = queryCategorizer.categorizeQuery();
        Assert.assertArrayEquals(new String[]{"sisi","sister"},
                queryKeeper.getPlusContain().toArray());

    }

    @Test
    public void testCategorizeForMinusQueries() {
        QueryKeeper queryKeeper = queryCategorizer.categorizeQuery();
        Assert.assertArrayEquals(new String[]{"brother","bro"},
                queryKeeper.getMinusContain().toArray());

    }


}

File: Phase05/Phase05/TestPhase05/QueryCategorizerTest.cs
using System;
using NSubstitute;
using Phase05;
using Xunit;

namespace TestPhase05
{
    public class QueryCategorizerTest
    {
        private readonly IQueryCategorizer _queryCategorizer;

        public QueryCategorizerTest()
        {
            _queryCategorizer = new QueryCategorizer();
        }
        
        [Fact]
        public void TestCategorize_ForNoSignQueries() {
            string[] queries = {"mom","dad","+sister","+sisi","-brother","-bro"};
            _queryCategorizer.CategorizeQueries(queries);
            Assert.assertArrayEquals(new String[]{"dad","mom"},
                _queryCategorizer.getQueryKeeper().getWithOutSign().toArray());

        }

        [Fact]
        public void testCategorizeForPlusQueries() {
            _queryCategorizer.categorizeQuery();
            Assert.assertArrayEquals(new String[]{"sisi","sister"},
                _queryCategorizer.getQueryKeeper().getPlusContain().toArray());

        }

        [Fact]
        public void testCategorizeForMinusQueries() {
            _queryCategorizer.categorizeQuery();
            Assert.assertArrayEquals(new String[]{"brother","bro"},
                _queryCategorizer.getQueryKeeper().getMinusContain().toArray());

        }
    }
}